packages/react-server/src/render/test/render.test.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import {createMockContext} from '@shopify/jest-koa-mocks';
+import withEnv from '@shopify/with-env';
+import {createRender, RenderContext} from '../render';
+import {LOGGER, noopLogger} from '../../logger';
+
+jest.mock('@shopify/sewing-kit-koa', () => ({
+  getAssets() {
+    return {
+      styles: () => Promise.resolve([]),
+      scripts: () => Promise.resolve([]),
+    };
+  },
+}));
+
+const meaningfulErrorMessage =
+  'Look, it broken. This is some meaningful error messsage.';
+const BrokenApp = function() {
+  throw new Error(meaningfulErrorMessage);
+};
+
+describe('createRender', () => {
+  it('response contains "My cool app"', async () => {
+    const myCoolApp = 'My cool app';
+    const ctx = createMockContext();
+    ctx.state[LOGGER] = noopLogger;
+
+    const renderFunction = createRender(() => <>{myCoolApp}</>);
+    await renderFunction(ctx as RenderContext);
+
+    expect(ctx.body).toContain(myCoolApp);
+  });
+
+  it('in development the body contains a meaningful error messages', () => {
+    withEnv('development', async () => {
+      const ctx = createMockContext();
+      ctx.state[LOGGER] = noopLogger;
+
+      const renderFunction = createRender(() => <BrokenApp />);
+      await renderFunction(ctx as RenderContext);
```

Comment:
This is fine as is, but you could wrap the `createMockContext` function with the missing properties from `RenderContext` and use that as the `ctx`.